Sync/SyncTelemetryUtils.swift

```diff
@@ -384,3 +384,72 @@ public struct SyncPing: SyncTelemetryPing {
         }
     }
 }
+
+public class GleanSyncOperationHelper {
+    public init () {}
+
+    public func start() {
+        GleanMetrics.Sync.syncUuid.generateAndSet()
+    }
+
+    public func end(_ result: SyncOperationResult) {
+        if let engineResults = result.engineResults.successValue {
+            engineResults.forEach { result in
+                let (name, status) = result
+                switch status {
+                case .completed(let stats):
+                    self.submitPingPerEngine(name, stats)
+                case .partial(let stats):
+                    self.submitPingPerEngine(name, stats)
+                case .notStarted(let reason):
+                    ()
+                    // failure_reason would be recorded here for the temp-<engine>-sync pings
+                }
+            }
+        } else if let failure = result.engineResults.failureValue {
+            // failure_reason would be recorded here for the temp-sync ping
```

Comment:
Yes, failure reason data will be helpful, and it might even be necessary? Well, at least in the redash query for the old data success records were distinguished from failure records based on whether a failure reason was present.